LaTeX/TeX.sublime-syntax

```diff
@@ -350,6 +247,122 @@ contexts:
     - include: brace-group-end
     - include: main
 
+###[ Registers ]################################################################
+
+  registers:
+    - match: (\\)({{registers}}){{endcs}}
+      captures:
+        1: punctuation.definition.backslash.tex
+        2: storage.type.tex
+      push: register-id
+
+  register-id:
+    - meta_scope: meta.register.tex
+    - match: \d+
+      scope: meta.number.integer.decimal constant.numeric.value.tex
+      set:
+        - tex-dimension-value
+        - tex-assignment
+    - match: (\\){{letter}}+
+      scope: support.function.general.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      set:
+        - tex-dimension-value
+        - tex-assignment
+    - include: paragraph-pop
+    - include: else-pop
+
+  register-allocations:
+    - match: (\\)new{{registers}}{{endcs}}
+      scope: keyword.declaration.register.tex storage.modifier.register.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      push: def-registername
+
+  def-registername:
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.tex
+      push: register-name-group-body
+    - match: (\\){{cmdname}}
+      scope: entity.name.constant.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+      pop: 1
+    - include: paragraph-pop
+    - include: else-pop
+
+  register-name-group-body:
+    - meta_scope: meta.group.brace.tex
+    - match: \}
+      scope: punctuation.definition.group.brace.end.tex
+      pop: 2
+    - match: (\\){{cmdname}}
+      scope: entity.name.constant.tex
+      captures:
+        1: punctuation.definition.backslash.tex
+
+
+###[ Dimensions ]###############################################################
+
+  tex-assignment:
+    - match: =
+      scope: keyword.operator.assignment.tex
+      pop: 1
+    - include: else-pop
+    - include: paragraph-pop
+
+
+  tex-dimension-value:
+    - match: (-?)(?:\d+(\.)\d*|(\.)\d+)
+      scope: constant.numeric.value.tex
+      captures:
+        1: keyword.operator.arithmetic.tex
```

Comment:
Nope. This scope has been rejected when numbers have been reworked. Please have a look in other syntaxes of this repo, which all use `keyword.operator`.

Reply:
This has all been discussed and agreements were made. No point in starting with it all over again.

Reply:
> We are talking about 04d5b6e and related commits, which really are a mistakes because some languages in some cases should scope them as signs, not operators.

@jrappen Whether the minus sign is actually treated as part of the number or as an operator is just an *implementation detail* of the language, with little relevance to users. The scoping as `keyword.operator` is done *by convention and for consistency* between syntaxes. Because all other syntaxes already use the `keyword.operator` scope, and it would be confusing if suddenly the number will be highlighted with the number color in just one or two syntaxes. And if you would want to introduce `constant.numeric.sign`, then all other syntaxes need to be revised as well, after a research of the particular low-level implementation of this for each language. Also, as @deathaxe already wrote, in many cases it is infeasible (requiring awkward lookbehinds or branching) or sheer impossible to apply the correct scoping dependant on surrounding context, linebreaks, etc., for example consider something like `int x = -42` versus `int x = 100 -42`.

If you start with this here, then in general many scopes in syntaxes need to be changed. For example in the MATLAB syntax, `true` and `false` are currently scoped as `constant.language`, as one would expect. But actually these are implemented as built-in functions in MATLAB. So if you want to be meticulous, they would have to be scoped `support.function`.
But that is just stupid from the highlighting point of view and for consistency with other syntaxes.

In some other languages, the type of `true` and `false` is a subtype of `Integer` (usually representing 1 and 0). So would you argue to scope it as `constant.numeric` in those langauges? I guess not.

Here is another example from the Python syntax. In Python, the `self` parameter name within a method body currently uses the `variable.language` scope. But following your logic, `self` would need to be removed from the special identifier names and scoped just with `meta.generic-name`, like any other random identifier name. Because in fact the name `self` is just a convention in Python, but with no special meaning to the language. If you want, you can use any other valid name for the first parameter of an instance method in Python, for example
```python
def __init__(this, foo, bar, baz):
    this.foo = foo
    this.bar = bar
    this.baz = baz
```
But I would argue that it is useful to keep the `variable.language` scope for `self`, because it is a pretty strong convention, and it improves the highlighting of Python code.

I could probably go on for hours with many other examples. But instead I hope that this comment has shown a reason for the current scoping of the minus sign and hopefully ends the discussions and potential skepticism about it once and for all.

Reply:
The argument of getting "conventional" highlighting instead of following the actual language rules is even more important for TeX, since almost every aspect of `TeX` can in principle be redefined by the user, down to the level of interpretation of single characters. 

What we are highlighting, in the end, is a set of such conventions (called a "format"), such as `LaTeX`, `BibTex`, and what we call `TeX` is really the `PlainTeX` format. But all of these are not new languages, but just define the state that the TeX interpreter is in after it has processed the commands in the format file.